app/index.js

```diff
@@ -112,8 +112,14 @@ loadAppStatePromise.then((initialImmutableState) => {
   CrashHerald.init(getSetting(SEND_CRASH_REPORTS, initialSettings))
 
   telemetry.setCheckpointAndReport('state-loaded')
-  if (getSetting(HARDWARE_ACCELERATION_ENABLED, initialSettings) === false) {
-    app.disableHardwareAcceleration()
+
+  // Check in case a user is upgrading with this setting.
+  // For non-legacy users, they will always have these settings in sync.
+  if (getSetting(HARDWARE_ACCELERATION_ENABLED, initialSettings) === false &&
+      app.getBooleanPref('hardware_acceleration_mode.enabled')) {
+    // This won't take effect until the next restart of the app, so some users
+    // may have to restart an extra time if UI doesn't work.
```

Comment:
Relating to this comment, I was going to relaunch here but ran into an issue. I posted this to investigate it further:
https://github.com/brave/browser-laptop/issues/12896